### ISSUE TITLE
jsx-shorthand-boolean: receive typeError stating it can't find rule when trying to lint

### ISSUE THREAD

**Issue by Tapialj** [eslint-stylistic/eslint-stylistic#1226]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

I am trying to set up my first stylistic config and all seems to work just fine except for the jsx-shorthand-boolean rule. I receive a TypeError stating that it could not find it in the plugin.

`TypeError: Key "rules": Key "@stylistic/jsx-shorthand-boolean": Could not find "jsx-shorthand-boolean" in plugin "@stylistic"`

I can comment the rule out and everything runs fine after.

Here is the line I am using:

`"@stylistic/jsx-shorthand-boolean": "error",`

And my esling.config.js:

```
import js from"@eslint/js";
import stylistic from"@stylistic/eslint-plugin";
import{ defineConfig } from"eslint/config";
import importPlugin from"eslint-plugin-import";
import reactPlugin from"eslint-plugin-react";
import globals from"globals";


export default defineConfig([
  {
    ignores: ["dist/**/*"],
  },
  js.configs.recommended,
  reactPlugin.configs.flat.recommended,
  {
    plugins: {
      import: importPlugin,
      "@stylistic": stylistic,
    },
  },
  {
    files: ["**/*.{js,mjs,cjs,jsx}"],
    // ...reactPlugin.configs.flat['jsx-runtime'],
    plugins: { js },
    extends: ["js/recommended"],
    languageOptions: {
      ecmaVersion: "latest",
      parserOptions: {
        ecmaFeatures: {
          jsx: true,
        },
      },
      globals: globals.browser,
    },
  },
  {
    settings: {
      react: {
        version: "detect",
      },
    },
  },
  {
    rules: {
      "no-unused-vars": "warn",
      "arrow-body-style": "off",
      "react/jsx-uses-react": "off",
      "react/react-in-jsx-scope": "off",
      "react/prop-types": "off",
      "@stylistic/array-element-newline": [
        "error",
        {
          consistent: true,
          multiline: true,
        },
      ],
      "@stylistic/arrow-parens": "error",
      "@stylistic/arrow-spacing": "error",
      "@stylistic/block-spacing": "error",
      "@stylistic/jsx-shorthand-boolean": "error",
    },
  },
]);
```

### Reproduction

Implement rule into eslint.config.js

### System Info

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (16) x64 AMD Ryzen 7 5800XT 8-Core Processor            
    Memory: 16.32 GB / 31.92 GB
  Binaries:
    Node: 25.8.2 - ~\Desktop\Other workspace\Projects\node_modules\.bin\node.CMD
    npm: 10.2.5 - ~\AppData\Roaming\npm\npm.CMD
  npmPackages:
    @stylistic/eslint-plugin: ^5.10.0 => 5.10.0 
    eslint: ^9.39.4 => 9.39.4
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
This rule was implemented in #1201, and first available in [v6.0.0-beta.5](https://github.com/eslint-stylistic/eslint-stylistic/releases/tag/v6.0.0-beta.5). You can try it now by upgrading to the beta version, or wait for the stable release.
